homeworks/hw2/Trie/Trie.cs

```diff
@@ -0,0 +1,163 @@
+﻿namespace TrieHomework;
+
+using System.Collections.Generic;
+
+// Trie data structure
+internal class Trie
+{
+    private class TrieNode
+    {
+        public Dictionary<char, TrieNode>? Children { get; set; } = new Dictionary<char, TrieNode>();
```

Comment:
Зачем Dictionary nullable? Он в конструкторе инициализируется не null-ом, и дальше вряд ли есть потребность его занулять.

И можно сделать
```suggestion
        public Dictionary<char, TrieNode>? Children { get; set; } = new();
```